numba/unicode.py

```diff
@@ -1361,6 +1361,43 @@ def impl(a):
     return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L9819-L9834    # noqa: E501
+@overload_method(types.UnicodeType, 'casefold')
+def unicode_casefold(data):
+    """Implements str.casefold()"""
+    def impl(data):
+        # https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L10782-L10791    # noqa: E501
```

Comment:
Presume you mean this as well? :
https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L9819-L9834

Reply:
Fixed!

Reply:
This SHA seems to have gone missing in the current patch set.

Reply:
I will add this SHA back. @stuartarchibald what do you think about returning `_do_casefold` directly from the implementation?
```
def unicode_casefold(data):
    """Implements str.casefold()"""
    return _do_casefold
```

Reply:
Ideal!

Reply:
Done!